src/click/decorators.py

```
@@ -339,22 +339,16 @@ def version_option(
         message = _("%(prog)s, version %(version)s")
 
     if version is None and package_name is None:
-        frame = inspect.currentframe()
-        assert frame is not None
-        assert frame.f_back is not None
-        f_globals = frame.f_back.f_globals if frame is not None else None
-        # break reference cycle
-        # https://docs.python.org/3/library/inspect.html#the-interpreter-stack
-        del frame
-
-        if f_globals is not None:
-            package_name = f_globals.get("__name__")
-
-            if package_name == "__main__":
-                package_name = f_globals.get("__package__")
-
-            if package_name:
-                package_name = package_name.partition(".")[0]
+        try:
+            f_globals = inspect.currentframe().frame.f_back.f_globals
+        except AttributeError:
+            raise AssertionError()
```

This pattern doesn't make sense. If there's an attribute error raise, just let that be raised, no need to silence it and turn it into an empty assertion error.

---

alright